src/ui/public/state_management/state_monitor.js

```diff
@@ -50,6 +55,17 @@ function stateMonitor(state, defaultState) {
   };
 
   return {
+    setDefault(defaultState) {
+      // update the originalState and apply ignoredProps
+      if (defaultState) {
+        setOriginalState(defaultState);
+        filterState(originalState);
+      }
+
+      // fire the change handler
+      dispatchChange();
```

Comment:
I would only expect a change notification changing the default state actually caused a state change. Could be as simple as calling `getStatus()` before and after then comparing before calling `dispatchChange()`.